src/aiortc/contrib/media.py

```diff
@@ -679,69 +804,114 @@ async def __run_track(self, track, context):
+                        elif track.kind == "lr_video":
+                            lr_frame_array, frame_index = await self.__lr_video_queue.get()
+                        if self.__display_option == "synthatic":
+                            if frame_index % self.__reference_update_freq == 0 and \
+                                    generator_type not in ['vpx', 'bicubic']:
+                                source_frame_array, source_keypoints, source_frame_index = await self.__source_queue.get()
+                                
+                                time_before_update = time.perf_counter()
+                                model.update_source(source_frame_index, source_frame_array, source_keypoints)
+                                time_after_update = time.perf_counter()
+                                self.__log_debug("Time to update source frame %s in receiver" \
+                                        " when receiving %s %s: %s",
+                                        source_frame_index, track.kind, frame_index, \
+                                        str(time_after_update - time_before_update))
+                                if self.__save_dir is not None:
+                                    pass
+                                    #np.save(os.path.join(self.__save_dir, 
+                                    #    'reference_frame_%05d.npy' % video_frame_index), source_frame_array)
+
+                            with concurrent.futures.ThreadPoolExecutor() as pool:
+                                if generator_type not in ['vpx', 'bicubic']:
+                                    self.__log_debug("Calling predict for frame %s with source frame %s",
+                                                frame_index, source_frame_index)
+                                before_predict_time = time.perf_counter()
+                                if track.kind == "keypoints":
+                                    predicted_target = await loop.run_in_executor(pool, \
+                                            model.predict, received_keypoints)
+                                elif track.kind == "lr_video":
+                                    if self.__prediction_type == "bicubic":
+                                        #predicted_target = await loop.run_in_executor(pool, resize_frame, lr_frame_array, frame_shape[0], device)
+                                        predicted_target = lr_frame.reformat(width=frame_shape[0], height=frame_shape[0],\
+                                                            interpolation='BICUBIC').to_rgb().to_ndarray()
+
+                                    else:
+                                        predicted_target = await loop.run_in_executor(pool, \
+                                                model.predict_with_lr_video, lr_frame_array)
+
+                            after_predict_time = time.perf_counter()
+                            if generator_type not in ['vpx', 'bicubic']:
```

Comment:
Doesn't seem bad to just report this regardless of the method - can remove this if

Reply:
removed